TLM/TLM/UI/Textures/SpeedLimitTextures.cs

```diff
@@ -20,196 +136,123 @@ public static class SpeedLimitTextures {
-            // TODO: Split loading here into dynamic sections, static enforces everything to stay in this ctor
-            RoadDefaults = new Dictionary<int, Texture2D>();
-            TexturesKmph = new Dictionary<int, Texture2D>();
-            TexturesMphUS = new Dictionary<int, Texture2D>();
-            TexturesMphUK = new Dictionary<int, Texture2D>();
+        public static readonly Dictionary<string, RoadSignTheme> Themes = new ();
 
-            IntVector2 sizeSquare = new IntVector2(200);
-            IntVector2 sizeUS = new IntVector2(200, 250);
+        private static RoadSignTheme activeTheme_ = null;
 
-            NoOverride = LoadDllResource(
-                resourceName: "SpeedLimits.NoOverride.png",
-                size: new IntVector2(200));
+        /// <summary>Names from <see cref="Themes"/> sorted.</summary>
+        public static readonly List<string> ThemeNames;
 
-            // Load shared speed limit signs for Kmph and Mph
-            // Assumes that signs from 0 to 140 with step 5 exist, 0 denotes no limit sign
-            for (var speedLimit = 0; speedLimit <= 140; speedLimit += 5) {
-                var resource = LoadDllResource(
-                    resourceName: $"SpeedLimits.Kmh.{speedLimit}.png",
-                    size: sizeSquare,
-                    mip: true);
-                TexturesKmph.Add(speedLimit, resource ? resource : TexturesKmph[5]);
-            }
+        public static RoadSignTheme ActiveTheme {
+            get {
+                if (activeTheme_ == null || !activeTheme_.AttemptedToLoad) {
+                    activeTheme_ = LoadCurrentTheme();
+                }
 
-            for (var speedLimit = 0; speedLimit <= 140; speedLimit += 5) {
-                var resource = LoadDllResource(
-                    resourceName: $"SpeedLimits.RoadDefaults.{speedLimit}.png",
-                    size: sizeSquare,
-                    mip: true);
-                RoadDefaults.Add(speedLimit, resource ? resource : RoadDefaults[5]);
+                return activeTheme_;
             }
+        }
 
-            // Signs from 0 to 90 for MPH
-            for (var speedLimit = 0; speedLimit <= 90; speedLimit += 5) {
-                // Load US textures, they are rectangular
-                var resourceUs = LoadDllResource(
-                    resourceName: $"SpeedLimits.Mph_US.{speedLimit}.png",
-                    size: sizeUS,
-                    mip: true);
-                TexturesMphUS.Add(speedLimit, resourceUs ? resourceUs : TexturesMphUS[5]);
-
-                // Load UK textures, they are square
-                var resourceUk = LoadDllResource(
-                    resourceName: $"SpeedLimits.Mph_UK.{speedLimit}.png",
-                    size: sizeSquare,
-                    mip: true);
-                TexturesMphUK.Add(speedLimit, resourceUk ? resourceUk : TexturesMphUK[5]);
-            }
+        public const string GERMAN_KM_SIGNS = "Kmph_Germany";
```

Comment:
why is this public and the rest not?

Reply:
It is used as default fallback theme in the Options General tab